mbed_build/_internal/config/config_modifiers.py

```diff
@@ -26,14 +26,16 @@ class SetConfigValue:
     value: Any
     help: Optional[str]
 
-    def __call__(self, config: "Config") -> "Config":
+    def __call__(self, config: "Config") -> None:
         """Mutate config by overwriting existing entry with new data."""
-        existing = config.settings.get(self.key, {})
-        override = {"value": self.value}
-        if self.help:
-            override["help"] = self.help
-        config.settings[self.key] = {**existing, **override}
-        return config
+        existing = config["settings"].get(self.key)
+        if existing:
+            config["settings"][self.key]["value"] = self.value
```

Comment:
We could reduce this down to `existing["value"] = self.value`? You're already holding a reference to `config["settings"][self.key]` in `existing`, right?

Reply:
Nice catch: https://github.com/ARMmbed/mbed-build/pull/19/commits/02748cd4799e5b839f22f5d9b3a63217e400c853